src/components/content/order/formElements/TextInput.tsx

```diff
@@ -8,23 +8,46 @@ import { EyeInvisibleOutlined, EyeTwoTone, InfoCircleOutlined } from '@ant-desig
 import { DeployParam, TextInputEventHandler } from './CommonTypes';
 import { DeployVariable } from '../../../../xpanse-api/generated';
 import React from 'react';
-
+import { Rule } from 'rc-field-form/lib/interface';
 export function TextInput({
     item,
     onChangeHandler,
 }: {
     item: DeployParam;
     onChangeHandler: TextInputEventHandler;
 }): React.JSX.Element {
+    const ruleItems: Rule[] = [{ required: item.mandatory }, { type: 'string' }];
+    let regExp: RegExp;
+    let minLength: number;
+    let maxLength: number;
+    const setValidator = (_: unknown, value: string) => {
+        if (value.length >= minLength && value.length <= maxLength && !regExp.test(value)) {
+            return Promise.reject(new Error(`The new password not match the pattern!${regExp}`));

```

Comment:
error text  is not correct. Better we execute each rule separately and return separate errors for minLength, maxLength and pattern.